contracts/src/v0.8/functions/dev/v1_X/FunctionsBilling.sol

```diff
@@ -256,21 +266,22 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
     bytes memory response,
     bytes memory err,
     bytes memory onchainMetadata,
-    bytes memory /* offchainMetadata TODO: use in getDonFee() for dynamic billing */
+    bytes memory /* offchainMetadata TODO: use in getDonFee() for dynamic billing */,
+    uint8 reportBatchSize
   ) internal returns (FunctionsResponse.FulfillResult) {
     FunctionsResponse.Commitment memory commitment = abi.decode(onchainMetadata, (FunctionsResponse.Commitment));
 
-    uint96 juelsPerGas = _getJuelsPerGas(tx.gasprice);
+    uint256 gasOverheadWei = (commitment.gasOverheadBeforeCallback + commitment.gasOverheadAfterCallback) * tx.gasprice;
+    uint256 l1FeeShareWei = ChainSpecificUtil._getCurrentTxL1GasFees(msg.data) / reportBatchSize;
     // Gas overhead without callback
-    uint96 gasOverheadJuels = juelsPerGas *
-      (commitment.gasOverheadBeforeCallback + commitment.gasOverheadAfterCallback);
+    uint96 gasOverheadJuels = _getJuelsFromWei(gasOverheadWei + l1FeeShareWei);
 
     // The Functions Router will perform the callback to the client contract
     (FunctionsResponse.FulfillResult resultCode, uint96 callbackCostJuels) = _getRouter().fulfill(
       response,
       err,
-      juelsPerGas,
-      gasOverheadJuels + commitment.donFee, // costWithoutFulfillment
+      _getJuelsPerGas(tx.gasprice),
```

Comment:
_getJuelsPerGas() is the exact same function. We don't need it any more, just use _getJuelsFromWei().